src/Raylib.php

```diff
@@ -504,6 +504,203 @@ public function getWorldToScreen2D(Types\Vector2 $position, Types\Camera2D $came
         return new Types\Vector2($vec->x, $vec->y);
     }
 
+    /**
+     * @psalm-suppress InvalidPassByReference
+     * @psalm-suppress MixedArgument
+     * @psalm-suppress MixedAssignment
+     * @psalm-suppress UndefinedPropertyFetch
+     */
+    public function imageCrop(Types\Image $image, Types\Rectangle $crop): void
+    {
+        $imageStruct = $image->toCData($this->ffi);
+        $this->ffi->ImageCrop(FFI::addr($imageStruct), $crop->toCData($this->ffi));
+
+        $image->data = $imageStruct->data;
```

Comment:
Moving data from the struct to the `Image` object happens multiple times in this class. Let's please extract this repeated logic into a private method of Raylib or a public method of `Image`.

Something like this:
```
$this->updateImage($image, $imageStruct);
```

Or this:
```
$image->updateFromStruct($imageStruct);
```

(names don't need to be the ones I posted here, just wanted to make my ideas clearer)